tests/potential/test_potential.py

```diff
@@ -60,38 +60,33 @@ def test_accessors(self):
         self.assertEqual(m['B','B']['energy'], None)
         self.assertEqual(m['B','B']['mass'], None)
 
-        m['A','B']['energy'] = 1.0
-        self.assertEqual(m['A','B']['energy'], 1.0)
-        self.assertEqual(m['A','B']['mass'], None)
-        self.assertEqual(m['A','A']['energy'], None)
-        self.assertEqual(m['A','A']['mass'], None)
-        self.assertEqual(m['B','B']['energy'], None)
-        self.assertEqual(m['B','B']['mass'], None)
-
-        m['A','B']['mass'] = 2.0
+        #test setting all key values at once
+        m['A','B'] = {'energy':1.0, 'mass':2.0}
         self.assertEqual(m['A','B']['energy'], 1.0)
         self.assertEqual(m['A','B']['mass'], 2.0)
         self.assertEqual(m['A','A']['energy'], None)
         self.assertEqual(m['A','A']['mass'], None)
         self.assertEqual(m['B','B']['energy'], None)
         self.assertEqual(m['B','B']['mass'], None)
 
-        m['A','A']['energy'] = 1.0
+        #test setting key values partially
+        m['A','A'] = {'energy':1.0}
         self.assertEqual(m['A','B']['energy'], 1.0)
         self.assertEqual(m['A','B']['mass'], 2.0)
```

Comment:
This is not really the behavior we want to have here, I think. When setting using the `=` (assignment) operator, we want to just set the values to exactly what's in the dict. If something is not in the dict, it should default to the `default` value specified in the constructor. If this is also not specified, it should be `None`. You will need to modify some of the methods above to get this to work.

Then, we will support this case of partial update with a new method later.

Reply:
I don't really understand what you're saying here, in your second sentence. Which `dict` should the values be set to? Should we just disallow partial reassignment?

Reply:
Yes, we don't want to allow "partial" reassignment by the `=` operator, just complete reassignment. I'll try to give an example.

Suppose we start with defaults
```py
m = CoefficientMatrix(types=('A',), params=('energy','mass'), default={'energy': 0.0, 'mass': 1.0})
```
If I assign on m, then I should get:
```py
>>> m['A','A'] = {'energy': 1.0, 'mass': 2.0}
>>> m['A','A']['energy'] == 1.0
True
>>> m['A','A']['mass'] == 2.0
True
```
If I started with the initial m and only assigned partially, I should leave the mass at its default:
```py
>>> m['A','A'] = {'energy': 1.0}
>>> m['A','A']['energy'] == 1.0
True
>>> m['A','A']['mass'] == 1.0
True
```
Now, if I combined the two (execute these blocks one after the other), I would expect the same behavior because the `=` assignment should completely overwrite the entry. I believe as implemented the mass stays at 2.0, and the last check would fail, right?

We will want to implement a separate `update` method (or something like that), which allows for partial assignment of parameters over existing ones.